src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -0,0 +1,414 @@
+                return (object)true;
+            else if (jsonElement.ValueKind == JsonValueKind.Number)
+            {
+                if (jsonElement.TryGetInt64(out long longValue))
+                    return longValue;
+                else if (jsonElement.TryGetInt32(out int intValue))
+                    return intValue;
+                else if (jsonElement.TryGetDecimal(out decimal decimalValue))
+                    return decimalValue;
+                else if (jsonElement.TryGetDouble(out double doubleValue))
+                    return doubleValue;
+                else if (jsonElement.TryGetUInt32(out uint uintValue))
+                    return uintValue;
+                else if (jsonElement.TryGetUInt64(out ulong ulongValue))
+                    return ulongValue;
+            }
+
+            return jsonElement.GetString();
+        }
+
+        internal Claim GetClaim(string key, string issuer)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            if (!RootElement.TryGetProperty(key, out JsonElement jsonElement))
+                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14304, key)));
+
+            return CreateClaimFromJsonElement(key, issuer, jsonElement);
+        }
+
+        internal static string GetClaimValueType(object obj)
+        {
+            if (obj == null)
+                return JsonClaimValueTypes.JsonNull;
+
+            var objType = obj.GetType();
+
+            if (objType == typeof(string))
+                return ClaimValueTypes.String;
+
+            if (objType == typeof(int))
+                return ClaimValueTypes.Integer;
+
+            if (objType == typeof(bool))
+                return ClaimValueTypes.Boolean;
+
+            if (objType == typeof(double))
+                return ClaimValueTypes.Double;
+
+            if (objType == typeof(long))
+            {
+                long l = (long)obj;
+                if (l >= int.MinValue && l <= int.MaxValue)
+                    return ClaimValueTypes.Integer;
+
+                return ClaimValueTypes.Integer64;
+            }
+
+            if (objType == typeof(DateTime))
+                return ClaimValueTypes.DateTime;
+
+            return objType.ToString();
+        }
+
+        internal string GetStringValue(string key)
+        {
+            if (RootElement.TryGetProperty(key, out JsonElement jsonElement) && jsonElement.ValueKind == JsonValueKind.String)
+                return jsonElement.GetString();
+
+            return string.Empty;
+        }
+
+        internal DateTime GetDateTime(string key)
+        {
+            if (!RootElement.TryGetProperty(key, out JsonElement jsonElement))
+                return DateTime.MinValue;
+
+            return EpochTime.DateTime(Convert.ToInt64(Math.Truncate(Convert.ToDouble(ParseTimeValue(key, jsonElement), CultureInfo.InvariantCulture))));
+        }
```

Comment:
I think this can be simplified to:

return EpochTime.DateTime(ParseTimeValue(key, jsonElement));

It should not need the conversion.